src/geo/data-providers/geojson/geojson-data-provider.js

```diff
@@ -15,23 +15,34 @@ var GeoJSONDataProvider = function (vectorLayerView, layerIndex) {
 // TODO: We can extract each "generator" to an individual file so that this file doesn't get too BIG
 GeoJSONDataProvider.prototype._dataGeneratorsForDataviews = {
   category: function (features, options) {
```

Comment:
from my point of view `features` is something totally internal that should not be passed. The dataprovider for the layer already knows where and how to get the data. The filterEnabled thing is something dataprovider needs in the same way windshaft needs to know it

agree on tests